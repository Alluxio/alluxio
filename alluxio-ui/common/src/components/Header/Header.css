/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */
/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */
.header .navbar {
  padding: 0; }
  .header .navbar .navbar-brand {
    padding: 0; }
    .header .navbar .navbar-brand .headerLogo .brand {
      font-size: 1.125rem;
      padding: 0.25em; }
      @media only screen {
        .header .navbar .navbar-brand .headerLogo .brand .brandImage img.brandSpin {
          height: 1.75em;
          width: 1.75em;
          -webkit-animation: brandSpin 20s ease-in-out infinite;
          -moz-animation: brandSpin 20s ease-in-out infinite;
          animation: brandSpin 20s ease-in-out infinite; } }
      .header .navbar .navbar-brand .headerLogo .brand .brandName {
        color: #fafafa;
        font-weight: 600;
        letter-spacing: 0.05em;
        transform: scaleY(0.8);
        image-rendering: crisp-edges; }

@keyframes brandSpin {
  0%, 90% {
    transform: rotateY(0deg); }
  95% {
    transform: rotateY(90deg); }
  100% {
    transform: rotateY(0deg); } }
  .header .navbar .tabs .nav-tabs {
    border: 0; }
    .header .navbar .tabs .nav-tabs .nav-link {
      padding: 0.25em 0.5em; }
      .header .navbar .tabs .nav-tabs .nav-link.active {
        background-color: #222;
        border-color: transparent; }
