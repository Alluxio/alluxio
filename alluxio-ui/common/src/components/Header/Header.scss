/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

@import '../../constants/scss/darkly/variables';

.header {
  .navbar {
    padding: 0;

    .navbar-brand {
      padding: 0;

      .headerLogo {
        .brand {
          padding: 0.25em;

          .brandImage {
            @media only screen {
              img.brandSpin {
                height: 1.75em;
                width: 1.75em;
                -webkit-animation: brandSpin 20s ease-in-out infinite;
                -moz-animation: brandSpin 20s ease-in-out infinite;
                animation: brandSpin 20s ease-in-out infinite;
              }
            }
          }

          .brandName {
            color: #fafafa;
            font-weight: 600;
            letter-spacing: 0.05em;
            transform: scaleY(0.8);
            image-rendering: crisp-edges;
          }
        }

        @keyframes brandSpin {
          0%, 90% {
            transform: rotateY(0deg);
          }
          95% {
            transform: rotateY(90deg);
          }
          100% {
            transform: rotateY(0deg);
          }
        }
      }
    }

    .tabs {
      margin-bottom: -0.9em;

      .nav-tabs {
        border: 0;

        .nav-link {
          padding: 0.25em 0.5em;

          &.active {
            background-color: $body-bg;
            border-color: transparent;
          }
        }

        .auto-refresh-button {
          margin-top: -1.125em;
          margin-right: 0.33em;
        }
      }
    }
  }
}
