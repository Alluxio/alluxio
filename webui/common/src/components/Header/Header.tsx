/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {faBell, faCheckSquare, faSquare} from '@fortawesome/free-regular-svg-icons'
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome'
import {Action, Location} from 'history';
import React from 'react';
import {Link} from 'react-router-dom';
import {
  Button,
  ButtonGroup,
  Collapse,
  Nav,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  NavItem,
  NavLink, Popover, PopoverBody
} from "reactstrap";

import {INavigationData, INavigationDataCallbackParameters} from '../../constants';
import logo from '../../images/alluxio-mark-tight-sm.svg';
import {isExternalLink} from '../../utilities';

import './Header.css';

export interface IHeaderProps {
  autoRefreshCallback?: (enable: boolean) => void;
  data: INavigationData[];
  callbackParameters?: INavigationDataCallbackParameters;
  history: any;
  newerVersionAvailable?: boolean
}

interface IHeaderState {
  isAutoRefreshing: boolean;
  isOpen: boolean;
  isPopoverOpen: boolean;
  pathname: string;
}

export class Header extends React.PureComponent<IHeaderProps, IHeaderState> {
  constructor(props: IHeaderProps) {
    super(props);

    const {history: {location: {pathname}}} = this.props;
    this.toggleHamburgerMenu = this.toggleHamburgerMenu.bind(this);
    this.toggleAutoRefresh = this.toggleAutoRefresh.bind(this);
    this.togglePopover = this.togglePopover.bind(this);
    this.renderNavItems = this.renderNavItems.bind(this);
    this.renderNewerVersionNotification = this.renderNewerVersionNotification.bind(this);
    this.closeHeaderOnClick = this.closeHeaderOnClick.bind(this);
    this.state = {isAutoRefreshing: false, isOpen: false, pathname, isPopoverOpen: false};
  }

  public componentDidMount() {
    const {history} = this.props;
    history.listen((loc: Location, action: Action) => {
      const {history: {location: {pathname}}} = this.props;
      this.setState((prevState) => (
        {...prevState, ...{pathname}})
      );
    });
  }

  public render(): JSX.Element {
    const {isAutoRefreshing, isOpen} = this.state;
    const {data} = this.props;
    return (
      <div className={'header'}>
        <Navbar className="headerNavigation navbar-dark bg-primary pb-0" expand="lg">
          <NavbarBrand tag={Link} to="/" href="/">
            <div className="headerLogo align-top d-inline-block ml-lg-2">
              <div className="brand">
                <div className="brandImage d-inline-block">
                  <img className="brandSpin" src={logo}/>
                </div>
                <div className="brandName d-inline-block align-middle pl-1">
                  ALLUXIO
                </div>
              </div>
            </div>
          </NavbarBrand>
          <NavbarToggler className="mr-1" onClick={this.toggleHamburgerMenu}/>
          <Collapse className={`mb-0 mt-2 d-lg-inline-flex justify-content-lg-center${isOpen ? '' : ' tabs'}`}
                    isOpen={isOpen} navbar={true}>
            <Nav tabs={!isOpen} vertical={isOpen}>
              {this.renderNavItems(data)}
            </Nav>
          </Collapse>
          <Collapse className={`d-lg-inline-flex justify-content-lg-end${isOpen ? '' : ' tabs'}`}
                    isOpen={isOpen} navbar={true}>
            <Nav tabs={!isOpen} vertical={isOpen}>
              <NavItem>
                {this.renderNewerVersionNotification()}
                <ButtonGroup className="auto-refresh-button mr-1">
                  <Button size="sm" color="secondary"
                          onClick={this.toggleAutoRefresh} active={isAutoRefreshing}>
                    <FontAwesomeIcon icon={isAutoRefreshing ? faCheckSquare : faSquare}/>&nbsp;
                    Auto Refresh
                  </Button>
                </ButtonGroup>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>
      </div>
    );
  }

  private renderNavItems(datas: INavigationData[]) {
    const {pathname} = this.state;
    const {callbackParameters} = this.props;
    return datas.map((data: INavigationData) => {
      const url = typeof data.url === 'function'
        ? (callbackParameters ? data.url(callbackParameters) : data.url({}))
        : data.url;
      return (
        <NavItem key={url}>
          <NavLink className="text-white" tag={isExternalLink(url) ? NavLink : Link} to={url} href={url}
                   active={pathname === url}
                   onClick={this.closeHeaderOnClick} {...data.attributes}>{data.innerText}</NavLink>
        </NavItem>
      )
    });
  }

  private renderNewerVersionNotification() {
    if (!this.props.newerVersionAvailable) return null;

    return (
      <ButtonGroup className="mr-1">
        <Button id="Popover1" size="sm" color="secondary">
          <FontAwesomeIcon icon={faBell}/>
        </Button>
        <Popover placement="bottom" isOpen={this.state.isPopoverOpen} target="Popover1" toggle={this.togglePopover} trigger="hover">
          <PopoverBody>
            <span>A new Alluxio version is available <a href="https://www.alluxio.io/download/">here</a>!</span>
          </PopoverBody>
        </Popover>
      </ButtonGroup>
    );
  }

  private closeHeaderOnClick() {
    this.setState({isOpen: false});
  }

  private toggleAutoRefresh() {
    const {autoRefreshCallback} = this.props;
    let {isAutoRefreshing} = this.state;
    isAutoRefreshing = !isAutoRefreshing;
    this.setState({isAutoRefreshing});
    if (autoRefreshCallback) {
      autoRefreshCallback(isAutoRefreshing);
    }
    this.closeHeaderOnClick();
  }

  private toggleHamburgerMenu() {
    this.setState({isOpen: !this.state.isOpen});
  }

  private togglePopover() {
    this.setState({isPopoverOpen: !this.state.isPopoverOpen});
  }
}
