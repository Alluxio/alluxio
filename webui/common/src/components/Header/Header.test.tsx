/*
 * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
 * (the "License"). You may not use this work except in compliance with the License, which is
 * available at www.apache.org/licenses/LICENSE-2.0
 *
 * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
 * either express or implied, as more fully set forth in the License.
 *
 * See the NOTICE file distributed with this work for information regarding copyright ownership.
 */

import {configure, mount, ReactWrapper, shallow, ShallowWrapper} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {createBrowserHistory, History, LocationState} from 'history';
import React from 'react';
import {StaticRouter} from 'react-router';

import {Header, IHeaderProps} from './Header';

configure({adapter: new Adapter()});

describe('Header', () => {
  let history: History<LocationState>;
  let props: IHeaderProps;

  beforeAll(() => {
    history = createBrowserHistory({keyLength: 0});
    history.push('/');
    props = {
      data: [],
      history
    };
  });

  describe('Shallow component', () => {
    let shallowWrapper: ShallowWrapper;

    beforeAll(() => {
      shallowWrapper = shallow(<Header {...props}/>);
    });

    it('Renders without crashing', () => {
      expect(shallowWrapper.length).toEqual(1);
    });

    it('Matches snapshot with newer version', () => {
      shallowWrapper.setProps({newerVersionAvailable: true});
      expect(shallowWrapper).toMatchSnapshot();
    });

    it('Matches snapshot', () => {
      expect(shallowWrapper).toMatchSnapshot();
    });
  });

  describe('React component', () => {
    let reactWrapper: ReactWrapper;
    let context = {};

    beforeAll(() => {
      reactWrapper = mount(
        <StaticRouter location={history.location} context={context}>
          <Header {...props}/>
        </StaticRouter>
      );
    });

    it('Renders without crashing', () => {
      expect(reactWrapper.length).toEqual(1);
    });

    it('Matches snapshot', () => {
      expect(reactWrapper).toMatchSnapshot();
    });
  });
});
