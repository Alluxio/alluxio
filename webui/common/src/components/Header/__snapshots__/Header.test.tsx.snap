// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header React component Matches snapshot 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location={
    Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    }
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      data={Array []}
      history={
        Object {
          "action": "PUSH",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 2,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="header"
      >
        <Navbar
          className="headerNavigation navbar-dark bg-primary pb-0"
          expand="lg"
          tag="nav"
        >
          <nav
            className="headerNavigation navbar-dark bg-primary pb-0 navbar navbar-expand-lg"
          >
            <NavbarBrand
              href="/"
              tag={[Function]}
              to="/"
            >
              <Link
                className="navbar-brand"
                href="/"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-brand"
                  href="/"
                  onClick={[Function]}
                >
                  <div
                    className="headerLogo align-top d-inline-block ml-lg-2"
                  >
                    <div
                      className="brand"
                    >
                      <div
                        className="brandImage d-inline-block"
                      >
                        <img
                          className="brandSpin"
                          src="alluxio-mark-tight-sm.svg"
                        />
                      </div>
                      <div
                        className="brandName d-inline-block align-middle pl-1"
                      >
                        ALLUXIO
                      </div>
                    </div>
                  </div>
                </a>
              </Link>
            </NavbarBrand>
            <NavbarToggler
              className="mr-1"
              onClick={[Function]}
              tag="button"
              type="button"
            >
              <button
                className="mr-1 navbar-toggler"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="navbar-toggler-icon"
                />
              </button>
            </NavbarToggler>
            <Collapse
              appear={false}
              className="mb-0 mt-2 d-lg-inline-flex justify-content-lg-center tabs"
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={350}
                unmountOnExit={false}
              >
                <div
                  className="mb-0 mt-2 d-lg-inline-flex justify-content-lg-center tabs collapse navbar-collapse"
                  style={Object {}}
                >
                  <Nav
                    tabs={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="nav nav-tabs"
                    />
                  </Nav>
                </div>
              </Transition>
            </Collapse>
            <Collapse
              appear={false}
              className="d-lg-inline-flex justify-content-lg-end tabs"
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Transition
                appear={false}
                enter={true}
                exit={true}
                in={false}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={350}
                unmountOnExit={false}
              >
                <div
                  className="d-lg-inline-flex justify-content-lg-end tabs collapse navbar-collapse"
                  style={Object {}}
                >
                  <Nav
                    tabs={true}
                    tag="ul"
                    vertical={false}
                  >
                    <ul
                      className="nav nav-tabs"
                    >
                      <NavItem
                        tag="li"
                      >
                        <li
                          className="nav-item"
                        >
                          <ButtonGroup
                            className="auto-refresh-button mr-1"
                            role="group"
                            tag="div"
                          >
                            <div
                              className="auto-refresh-button mr-1 btn-group"
                              role="group"
                            >
                              <Button
                                active={false}
                                color="secondary"
                                onClick={[Function]}
                                size="sm"
                                tag="button"
                              >
                                <button
                                  aria-label={null}
                                  className="btn btn-secondary btn-sm"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <FontAwesomeIcon
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon={
                                      Object {
                                        "icon": Array [
                                          448,
                                          512,
                                          Array [],
                                          "f0c8",
                                          "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z",
                                        ],
                                        "iconName": "square",
                                        "prefix": "far",
                                      }
                                    }
                                    inverse={false}
                                    listItem={false}
                                    mask={null}
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    title=""
                                    transform={null}
                                  >
                                    <svg
                                      aria-hidden="true"
                                      className="svg-inline--fa fa-square fa-w-14 "
                                      data-icon="square"
                                      data-prefix="far"
                                      focusable="false"
                                      role="img"
                                      style={Object {}}
                                      viewBox="0 0 448 512"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <path
                                        d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
                                        fill="currentColor"
                                        style={Object {}}
                                      />
                                    </svg>
                                  </FontAwesomeIcon>
                                    Auto Refresh
                                </button>
                              </Button>
                            </div>
                          </ButtonGroup>
                        </li>
                      </NavItem>
                    </ul>
                  </Nav>
                </div>
              </Transition>
            </Collapse>
          </nav>
        </Navbar>
      </div>
    </Header>
  </Router>
</StaticRouter>
`;

exports[`Header Shallow component Matches snapshot 1`] = `
<div
  className="header"
>
  <Navbar
    className="headerNavigation navbar-dark bg-primary pb-0"
    expand="lg"
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag={[Function]}
      to="/"
    >
      <div
        className="headerLogo align-top d-inline-block ml-lg-2"
      >
        <div
          className="brand"
        >
          <div
            className="brandImage d-inline-block"
          >
            <img
              className="brandSpin"
              src="alluxio-mark-tight-sm.svg"
            />
          </div>
          <div
            className="brandName d-inline-block align-middle pl-1"
          >
            ALLUXIO
          </div>
        </div>
      </div>
    </NavbarBrand>
    <NavbarToggler
      className="mr-1"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      className="mb-0 mt-2 d-lg-inline-flex justify-content-lg-center tabs"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        tabs={true}
        tag="ul"
        vertical={false}
      />
    </Collapse>
    <Collapse
      appear={false}
      className="d-lg-inline-flex justify-content-lg-end tabs"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        tabs={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <ButtonGroup
            className="mr-1"
            role="group"
            tag="div"
          >
            <Button
              color="secondary"
              id="Popover1"
              size="sm"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0f3",
                      "M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z",
                    ],
                    "iconName": "bell",
                    "prefix": "far",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </Button>
            <Popover
              isOpen={false}
              placement="bottom"
              placementPrefix="bs-popover"
              target="Popover1"
              toggle={[Function]}
              trigger="hover"
            >
              <PopoverBody
                tag="div"
              >
                <span>
                  A new Alluxio version is available 
                  <a
                    href="https://www.alluxio.io/download/"
                  >
                    here
                  </a>
                  !
                </span>
              </PopoverBody>
            </Popover>
          </ButtonGroup>
          <ButtonGroup
            className="auto-refresh-button mr-1"
            role="group"
            tag="div"
          >
            <Button
              active={false}
              color="secondary"
              onClick={[Function]}
              size="sm"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0c8",
                      "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z",
                    ],
                    "iconName": "square",
                    "prefix": "far",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
                Auto Refresh
            </Button>
          </ButtonGroup>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;

exports[`Header Shallow component Matches snapshot with newer version 1`] = `
<div
  className="header"
>
  <Navbar
    className="headerNavigation navbar-dark bg-primary pb-0"
    expand="lg"
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag={[Function]}
      to="/"
    >
      <div
        className="headerLogo align-top d-inline-block ml-lg-2"
      >
        <div
          className="brand"
        >
          <div
            className="brandImage d-inline-block"
          >
            <img
              className="brandSpin"
              src="alluxio-mark-tight-sm.svg"
            />
          </div>
          <div
            className="brandName d-inline-block align-middle pl-1"
          >
            ALLUXIO
          </div>
        </div>
      </div>
    </NavbarBrand>
    <NavbarToggler
      className="mr-1"
      onClick={[Function]}
      tag="button"
      type="button"
    />
    <Collapse
      appear={false}
      className="mb-0 mt-2 d-lg-inline-flex justify-content-lg-center tabs"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        tabs={true}
        tag="ul"
        vertical={false}
      />
    </Collapse>
    <Collapse
      appear={false}
      className="d-lg-inline-flex justify-content-lg-end tabs"
      enter={true}
      exit={true}
      in={false}
      isOpen={false}
      mountOnEnter={false}
      navbar={true}
      onEnter={[Function]}
      onEntered={[Function]}
      onEntering={[Function]}
      onExit={[Function]}
      onExited={[Function]}
      onExiting={[Function]}
      tag="div"
      timeout={350}
      unmountOnExit={false}
    >
      <Nav
        tabs={true}
        tag="ul"
        vertical={false}
      >
        <NavItem
          tag="li"
        >
          <ButtonGroup
            className="mr-1"
            role="group"
            tag="div"
          >
            <Button
              color="secondary"
              id="Popover1"
              size="sm"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0f3",
                      "M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z",
                    ],
                    "iconName": "bell",
                    "prefix": "far",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </Button>
            <Popover
              isOpen={false}
              placement="bottom"
              placementPrefix="bs-popover"
              target="Popover1"
              toggle={[Function]}
              trigger="hover"
            >
              <PopoverBody
                tag="div"
              >
                <span>
                  A new Alluxio version is available 
                  <a
                    href="https://www.alluxio.io/download/"
                  >
                    here
                  </a>
                  !
                </span>
              </PopoverBody>
            </Popover>
          </ButtonGroup>
          <ButtonGroup
            className="auto-refresh-button mr-1"
            role="group"
            tag="div"
          >
            <Button
              active={false}
              color="secondary"
              onClick={[Function]}
              size="sm"
              tag="button"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f0c8",
                      "M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z",
                    ],
                    "iconName": "square",
                    "prefix": "far",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
                Auto Refresh
            </Button>
          </ButtonGroup>
        </NavItem>
      </Nav>
    </Collapse>
  </Navbar>
</div>
`;
